components/todolist.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from "react";
+import { FlatList, StyleSheet, View } from "react-native";
+import Header from "./header";
+import TaskList from "./taskList";
+import TaskInput from "./taskInput";
+
+class ToDoList extends Component {
+  state = {
+    currentId: 3,
+    tasks: [
+      { id: 1, description: "Call mum", isCompleted: false },
+      { id: 2, description: "Walk dog", isCompleted: false },
+      { id: 3, description: "Do laundry", isCompleted: false }
+    ]
+  };
+
+  handleDelete = taskId => {
+    const tasks = this.state.tasks.filter(t => t.id !== taskId);
+    this.setState({ tasks: tasks });
+  };
+
+  handleAddTask = description => {
+    this.state.currentId++;
+    const tasks = [
+      ...this.state.tasks,
+      { id: String(this.state.currentId), description: description }
+    ];
+    this.setState({ tasks: tasks });
+  };
+
+  handleDoTask = taskId => {
+    const tasks = [...this.state.tasks];
+    const index = tasks.findIndex(t => t.id == taskId);
+    tasks[index] = { ...tasks[index] };
+    tasks[index].isCompleted = !tasks[index].isCompleted;
```

Comment:
will be easier like this
` handleDoTask = taskId => {`
`    cons tasks = this.state.map((task, index)=> index == taskId ? {...task, isCompleted: !task.isComplemted} : task)`
`    this.setState({tasks});`
`}`

}